### ISSUE TITLE
how to calculate “img_norm_cfg” for custom dataset

### ISSUE THREAD

**Issue by donglee-afar** [open-mmlab/mmdetection#354]
“img_norm_cfg” is a  parameter in config file,how to calculate “img_norm_cfg” for custom dataset?
Could you provide a script file as a reference, thank you！

**Comment by hellock** [comment]
Generally you don't need to calculate `img_norm_cfg`. It is determined by the pretrained model, and you can just keep it the same if you use the model from pytorch model zoo.

**Comment by donglee-afar** [comment]
thank you！

**Comment by ximitiejiang** [comment]
hello @hellock 
I have similar confusion, I saw SSD detector with vgg16-caffe pretrained model with mean=[123.675, 116.28, 103.53], std=[1, 1, 1] for both voc & coco dataset, and vgg16-pytorch pretrained model usually use mean=[0.485, 0.456, 0.406]，std=[0.229, 0.224, 0.225], are they both pretrained on Imagenet, why they are totally different for same dataset?
I am sorry i do not familiar with caffe2, could you give a little more detail explanation  or reference about this topic?

**Comment by hellock** [comment]
@ximitiejiang mean and std is determined by the pretrained model. The pretrained model used in SSD is not trained by us or pytorch model zoo, and we have to keep the same mean and std used in pretraining.

**Comment by ximitiejiang** [comment]
Hi, @hellock, 
my understanding is if we calculate norm params with img/255, the mean/std should be mean=[0.485, 0.456, 0.406]，std=[0.229, 0.224, 0.225], and if we calculate with img , the mean/std should be mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], and for caffe model, mean=[123.675, 116.28, 103.53], std=[1, 1, 1], only need to transform to (img-mean)?

**Comment by hellock** [comment]
@ximitiejiang Yes that depends on the image preprocessing of pretraining, some people prefer to normalize the image to [0, 1] first and some don't. We follow pytorch's mean and std when using pretrained models of pytorch model zoo, and the model owner's mean and std when using others.

**Comment by ximitiejiang** [comment]
ok, thanks!

**Comment by callzhang** [comment]
My question here is "How" to calculate the `img_norm_cfg` for custom dataset fine tuning. Thanks.

**Comment by callzhang** [comment]
I end up wrote this script. Not sure if this is the right way.
```
from torch.utils.data import Dataset, DataLoader
from torch import Tensor
class MyDataset(Dataset):
    def __init__(self, image_list):
        assert isinstance(image_list, list)
        self.image_list = image_list
    
    def __getitem__(self, index):
        path = base+img_dir+self.image_list[index]
        image = Image.open(path).convert("RGB")
        img_arr = np.asarray(image, dtype=np.float32)
        assert img_arr.dtype == np.float32
        return img_arr

    def __len__(self):
        return len(self.image_list)

dataset = MyDataset(img_table.dropna().img.to_list())
loader = DataLoader(
    dataset,
    batch_size=100,
    num_workers=1,
    shuffle=False
)



mean = Tensor([0,0,0])
std = Tensor([0,0,0])
n_samples= 0
for data in tqdm(loader):
    batch_samples = data.size(0)
    data2 = data.view(-1, 3)
    mean += data2.mean(0)
    std += data2.std(0)
    n_samples += 1

mean /= n_samples
std /= n_samples
```
